### PR TITLE
Tone updates

### PR DESCRIPTION
**Pull request by ranj063**
This set of patches adds the changes required to add the virtual FE DAI link information in the
tone pipeline topology.

### CODE REVIEW

File: topology/m4/dai.m4 [hunk 1]
```diff
@@ -188,4 +188,14 @@ define(`DAI_ADD',
 `include($1)'
 )
 
+dnl VFE_LINK(link_name, cpu_dai_name, platform_name)
```
Comment (lgirdwood) [inline topology/m4/dai.m4]: Why do we need this ? it looks like we are defining a BE.

File: topology/m4/tone.m4 [hunk 2]
```diff
@@ -18,11 +18,13 @@ define(`W_TONE',
...
 `	tokens "sof_tone_tokens"'
 `	tuples."word" {'
-`		SOF_TKN_TONE_SAMPLE_RATE'		ifdef(TONE_SAMPLE_RATE, STR(TONE_SAMPLE_RATE), "48000")
+`		SOF_TKN_TONE_SAMPLE_RATE'		ifdef(`TONE_SAMPLE_RATE', STR(TONE_SAMPLE_RATE), "48000")
 `	}'
 `}'
 `SectionData."'N_TONE($1)`_data_w" {'
 `	tuples "'N_TONE($1)`_tuples_w"'
+`}'
+`SectionData."'N_TONE($1)`_tone_data_w" {'
```
Comment (lgirdwood) [inline topology/m4/tone.m4]: I'm not sure why we need to define a BE here when the routing will describe all the BEs that tone is connected too. I was expecting to see more a a flag that would say to create a VFE (where any strings for VFE would come from w
tone widget)

File: topology/m4/dai.m4 (hunk 1 above)
Comment (ranj063) [reply topology/m4/dai.m4]: @lgirdwood I cant get the dpcm_runtime_update() function to startup the codec without specifying the CPU dai name in the virtual FE. In the case of the up2 with hifiberry , this is "SSP5 Pin". So I thought the topology would be the best place to define it. Is there any other way I can get the cpu dai name while loading the siggen widget?

Comment (lgirdwood) [reply topology/m4/dai.m4]: ok, the tone should be routed using DAPM to the DAI or any other endpoint. So DAPM alone should route tone to SSP5. The VFE should have no string or other references (unless through DAPM) to any BE or BEs.  You may need to debug why the VFE needs a BE string reference if this does not work alone.

**Comment by ranj063** [comment]
closing this one. Will open a new one based on the latest tone kernel updates later.

**Pull request closed without merging**